internal/common/buffer_util_test.go

```diff
@@ -58,7 +58,7 @@ func TestReadBool(t *testing.T) {
 	buf := []byte{0, 0, 0}
 	WriteBool(buf, 0, true)
 	WriteBool(buf, 2, true)
-	if ReadBool(buf, 0) != true && ReadBool(buf, 2) != true {
+	if ReadBool(buf, 0) != true || ReadBool(buf, 2) != true {
```

Comment:
This should be `OR`. No need to change.

Reply:
If one of them is not `true`. It should fail.

Reply:
That is why it should be 'or'. Currently it is 'and' so if one of them fails it doesnt fail

Reply:
I am confused. Ignore this.